src/framework/execution/dependencyGraph.ts

```diff
@@ -8,22 +8,22 @@ import {
   MemoryDataStore,
 } from './jobState';
 import {
-  IntegrationExecutionContext,
-  IntegrationStep,
-  IntegrationStepExecutionContext,
+  Step,
   IntegrationStepResult,
-  IntegrationStepResultStatus,
-  IntegrationStepStartStates,
+  StepResultStatus,
+  StepStartStates,
+  ExecutionContext,
+  StepExecutionContext,
 } from './types';
 
 /**
  * This function accepts a list of steps and constructs a dependency graph
  * using the `dependsOn` field from each step.
  */
-export function buildStepDependencyGraph(
-  steps: IntegrationStep[],
-): DepGraph<IntegrationStep> {
-  const dependencyGraph = new DepGraph<IntegrationStep>();
+export function buildStepDependencyGraph<
+  TStepExecutionContext extends StepExecutionContext
+>(steps: Step<TStepExecutionContext>[]): DepGraph<Step<TStepExecutionContext>> {
+  const dependencyGraph = new DepGraph<Step<TStepExecutionContext>>();
```

Comment:
FWIW, I'm not a huge fan of `TStepExecutionContext` over just `T` in many cases, where there aren't but a few lines of code (a good practice for most functions!), but it certainly is clear.